stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/handlers/OnBackPressedHandler.kt

```diff
@@ -27,14 +26,13 @@ public fun SystemBackPressedHandler(
     onBackPressed: () -> Unit,
 ) {
     if (LocalInspectionMode.current) return
-
-    val backPressedDispatcher =
-        (LocalLifecycleOwner.current as ComponentActivity).onBackPressedDispatcher
+    val backPressedDispatcherOwner = LocalOnBackPressedDispatcherOwner.current
+    val dispatcher = backPressedDispatcherOwner?.onBackPressedDispatcher ?: return
 
     val callback = buildBackPressedCallback(isEnabled, onBackPressed)
 
-    DisposableEffect(backPressedDispatcher) {
-        backPressedDispatcher.addCallback(callback)
+    DisposableEffect(backPressedDispatcherOwner) {
+        dispatcher.addCallback(callback)
 
         onDispose {
             callback.remove()
```

Comment:
I think this is incorrect. If the whole function is recomposed for some reason, the callback will not be updated. `rememberUpdatedState` should be used instead:
https://developer.android.com/jetpack/compose/side-effects#disposableeffect